### DISCUSSION TITLE
Downloaded Twitter images are capped in resolution

### DISCUSSION THREAD

**Discussion by Technotron21** [TumblThreeApp/TumblThree#315] [General]
**Describe the bug**
I've noticed since the Twitter crawler was introduced that many of the higher resolution images end up having the limit of 2048 for length/width/both. Kind of similar to the issue the Tumblr crawler was having, but this one is more consistent from picture to picture.

**To Reproduce**
Grab a copy of a hi res image off the website and compare to the copy the crawler produces.

**Expected behavior**
The image from the crawler gets to 2048 maximum at either length, width, or both.

**Desktop (please complete the following information):**
 - TumblThree version: 2.5.0.249 (though prevalent since 2.0.0)
 - OS: Windows 10 Pro 21H1

**Comment by thomas694** [comment]
Why are you breaking your silence after so many months? /s 😉 

🙂 Thanks for bringing it to our attention. We'll have a look at it.
Can you give us an example post (of an high-res image) that is safe to post here (the more recent the better)?

**Comment by Technotron21** [comment]
Sorry, been a bit busy. I would've posted yesterday, but the limit issue was still in effect.

Here are two images that while not the most the site can offer, does show the issue fairly well:
https://pbs.twimg.com/media/FGJtUq_XwAEtNdZ?format=jpg&name=4096x4096 (website copy)
https://i.imgur.com/orvPgY2.jpg (downloaded copy)

https://pbs.twimg.com/media/FJEIq6_VcAABBVz?format=jpg&name=4096x4096 (website copy)
https://i.imgur.com/5k58TCp.jpg (downloaded copy)

**Comment by thomas694** [comment]
Thanks for the examples.

**Comment by Technotron21** [comment]
Also, I don't know if this is related at all, but I did notice downloaded video thumbnails being higher resolution than the actual video files themselves. I'm not sure if it's a genuine glitch or an accidental up-res on the crawler's end, but I thought I should mention it.

**Comment by thomas694** [comment]
As you found out there was a problem with very high resolution images. Now there exists a new image category "best" for downloading them.

Of course, a thumbnail image in a higher resolution than the video doesn't make much sense. The app just always requested a high resolution image and Twitter kindly delivered (an upscaled) one. Only the original thumbnail is downloaded now.

**Comment by thomas694** [comment]
The issue has been fixed and closed. You can still comment. Feel free to ask for reopening the issue if needed.
